runtime/src/main/scala/akka/grpc/GrpcProtocol.scala

```diff
@@ -121,7 +119,13 @@ object GrpcProtocol {
    * Detects which gRPC protocol variant is indicated by a mediatype.
    * @return a [[GrpcProtocol]] matching the request mediatype if known.
    */
-  def detect(mediaType: jmodel.MediaType): Option[GrpcProtocol] = protocols.find(_.mediaTypes.contains(mediaType))
+  def detect(mediaType: jmodel.MediaType): Option[GrpcProtocol] = mediaType.subType match {
+    // FIXME: do we need to check mainType?
+    case "grpc" | "grpc+proto"                   => Some(GrpcProtocolNative)
```

Comment:
No need for relatively complex collection logic when we can just match on strings.